rotkehlchen/chain/evm/decoding/balancer/v2/balances.py

```diff
@@ -0,0 +1,83 @@
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from eth_utils import to_checksum_address
+
+from rotkehlchen.accounting.structures.balance import Balance, BalanceSheet
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.ethereum.interfaces.balances import BalancesSheetType, ProtocolWithBalance
+from rotkehlchen.chain.evm.decoding.balancer.constants import (
+    CHAIN_ID_TO_BALANCER_API_MAPPINGS,
+    CPT_BALANCER_V2,
+    USER_GET_POOL_BALANCES_QUERY,
+)
+from rotkehlchen.chain.evm.decoding.balancer.utils import query_balancer_api
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import EvmTokenKind
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.decoder import EVMTransactionDecoder
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.types import ChecksumEvmAddress
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class Balancerv2Balances(ProtocolWithBalance):
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            tx_decoder: 'EVMTransactionDecoder',
+    ):
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            tx_decoder=tx_decoder,
+            counterparty=CPT_BALANCER_V2,
+            deposit_event_types={(HistoryEventType.DEPOSIT, HistoryEventSubType.DEPOSIT_ASSET)},
+        )
+
+    def query_balances(self) -> 'BalancesSheetType':
```

Comment:
isn't this function very similart ot v1 balances? Can't you just simply abstract it somehow?